desmume/src/frontend/posix/gtk/main.cpp

```diff
@@ -2087,10 +2097,16 @@ static void GraphicsSettingsDialog(GSimpleAction *action, GVariant *parameter, g
 		default:
 			break;
 		}
-		gpu_scale_factor = gtk_combo_box_get_active(wGPUScale)+1;
+		gpu_scale_factor = gtk_spin_button_get_value(wGPUScale);
+		if(gpu_scale_factor < GPU_SCALE_FACTOR_MIN)
+			gpu_scale_factor = GPU_SCALE_FACTOR_MIN;
+		if(gpu_scale_factor > GPU_SCALE_FACTOR_MAX)
+			gpu_scale_factor = GPU_SCALE_FACTOR_MAX;
```

Comment:
might want to save the normalized value back into the spin button in case it exceeds the limits